tests/e2e/specs/myOrders_spec.js

```diff
@@ -0,0 +1,51 @@
+describe('my orders', () => {
+  const customer = {
+    firstName: 'Charlie',
+    lastName: 'Bucket',
+    email: 'charlie.bucket+ci@commercetools.com',
+    password: 'p@ssword',
+  };
+
+  const cartDraft1 = {
+    customerEmail: 'charlie.bucket+ci@commercetools.com',
+    currency: 'EUR',
+    lineItems: {
+      sku: 'M0E20000000DPZ0',
+    },
+    shippingAddress: {
+      country: 'DE',
+    },
+  };
+
+  const cartDraft2 = {
+    customerEmail: 'charlie.bucket+ci@commercetools.com',
+    currency: 'EUR',
+    lineItems: {
+      sku: 'M0E20000000DLPH',
+    },
+    shippingAddress: {
+      country: 'DE',
+    },
+  };
+
+  before(() => {
+    cy.login(customer);
+  });
+
+  it('shows my orders', () => {
+    cy.createMyOrder(cartDraft1);
+    cy.createMyOrder(cartDraft2);
+    cy.get('[data-test=my-orders-button]').click();
+    cy.reload();
+    cy.url().should('include', '/user/orders');
+    cy.get('[data-test=order-date]')
+      .should('exist');
+    cy.get('[data-test=order-list]')
+      .should('have.length', 2)
+      .eq(0)
+      .contains(/^\s*368,75\s€\s*$/);
+    cy.get('[data-test=order-list]')
+      .eq(1)
+      .contains(/^\s*372,50\s€\s*$/);
```

Comment:
You don't have to check both orders. Checking too few is bad, but checking too much affects performance (and e2e will become veeery slow in the future, we have to avoid over-testing as much as we can).

In general, what I would do is:
- check we have 2 orders (i.e. `should have length 2`)
- get one of the orders (i.e. `eq(0)`)
- for this order only, check that every field (order number, date, price) matches what I expect to see (with regular expressions or exact text if we have it)

This way we don't under- or over-test.

Regarding Payment status and Shipping status we have to display it as well in a certain way and test them, but I have created an issue for that: https://github.com/commercetools/sunrise-spa/issues/107 so we'll have to take care of that in the future.